xarray/core/common.py

```diff
@@ -418,6 +418,9 @@ def _maybe_promote(dtype):
         # convert to floating point so NaN is valid
         dtype = float
         fill_value = np.nan
+    elif np.issubdtype(dtype, complex):
+        #explicit cast of np.nan is probibly not necassary
+        fill_value = np.complex(np.nan)
```

Comment:
Actually, let's use: `np.nan + np.nan * 1j`. Otherwise, the NaN is only done for real part of the number:
```
In [19]: complex(np.nan)
Out[19]: (nan+0j)
```

I'm not sure this actually matters but it can't hurt